#161124305 Users should be able to view all articles by category

emekafredy opened the pull request:
#### What does this PR do?
- Users should be able to view articles by the categories

#### Description of Task to be completed?
- implement feature to view all articles in a category
- design page to view all categories available in the app
- implement feature to view all categories


#### How should this be manually tested?
- clone the branch repo ``` ft-view-articles-by-category-161124305 ```
- cd into the directory
- run ``` npm install ```
- run ``` npm start ```
- click on any of the categories to view collections

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[161124305](https://www.pivotaltracker.com/story/show/161124305)

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-10-12 at 7 00 43 pm" src="https://user-images.githubusercontent.com/29521319/46886706-8d1e9800-ce53-11e8-9a56-9124d415dc22.png">
<img width="1440" alt="screen shot 2018-10-12 at 7 00 56 pm" src="https://user-images.githubusercontent.com/29521319/46886716-94de3c80-ce53-11e8-84ee-7328e800df8d.png">
<img width="1440" alt="screen shot 2018-10-12 at 7 01 07 pm" src="https://user-images.githubusercontent.com/29521319/46886720-99a2f080-ce53-11e8-8f27-1d8ee8600c6f.png">
<img width="398" alt="screen shot 2018-10-12 at 7 03 45 pm" src="https://user-images.githubusercontent.com/29521319/46886732-9f003b00-ce53-11e8-8083-4d7f0a15a5e2.png">
<img width="430" alt="screen shot 2018-10-12 at 7 03 32 pm" src="https://user-images.githubusercontent.com/29521319/46886736-a32c5880-ce53-11e8-9d3d-57fd43ff8722.png">

#### Questions:

emekafredy committed:
ft(articleCategories): implement feature for article categories view

- implement feature to view all articles in a category
- design page to view all categories available in the app
- implement feature to view all categories

[Delivers #161124305]

emmaadesile commented on src/containers/Categories/Categories.js:
```diff
@@ -0,0 +1,91 @@
+    }
+  
+    const { data } = categories;
+    console.log('DATA', data);
```
Good job destructuring the data from the categories. Is there a reason why you are logging the DATA to the console?

emekafredy replied:
I forgot to remove this. Thanks for bringing it to my notice

emmaadesile commented on src/containers/Categories/SingleCategory.js:
```diff
@@ -0,0 +1,121 @@
+
+    if (category && !category.isLoading) {
+      data = category.data;
+      console.log('DATA', data);
```
You should also remove the `console.log statement` here

emekafredy replied:
I will also do this. Thanks

emmaadesile commented on src/containers/Categories/SingleCategory.js:
```diff
@@ -0,0 +1,121 @@
+              const snippet = description;
+              const details = {
+                author: `${author.firstName} ${author.lastName}`,
+                timeToRead: `${timeToRead} min Read`,
```
line 84. `${timeToRead} min Read` R in Read should be in lowercase.

emekafredy replied:
Thanks for the feedback. I will correct this

CEOehis commented on src/store/initialState.js:
```diff
@@ -10,6 +10,10 @@ const initialState = {
+  articleCategory: {
+    category: null,
+    categories: [],
   }
```
I think the whole idea of the `initialState.js` file is to give a clear view of what the app state looks like when loaded. In that light you can make the values in your initial state more descriptive of the value it would finally hold. For instance `category: null` does not let one know what the shape of `category` will be like later on.

Also you have `categories: []` but in the `reducers/category.js` file you set it to be an object with shape 
```
      const categories = {
        isLoading: false,
        isError: false,
        data: action.payload
      };
```

I think you can declare this shape in the initial state so anyone looking through the code knows what the shape/interface of `categories` looks like. Same thing goes for the `category` state.

emekafredy replied:
Okay. I will include these in the initial state

emekafredy committed:
nitpicks(articleCategories): implement requested changes

- clean up code by removing every console.log fn
- initialState.js should have all the contents of the category's initial state

CEOehis reviewed (approved):
LGTM

johnkingzy reviewed (approved):
Looks good to me
